lib/resources/areas.js

```diff
@@ -1,125 +1,77 @@
-var Areas;
-var request = require('request');
-var fs      = require('fs');
-
-function Areas (config) {
-  this.uri = config.endpoint + 'areas';
-  this.apiKey = config.apiKey;
-  this.userAgent = config.userAgent;
-  this.apiVersion = config.apiVersion;
-  return this;
-}
-
-Areas.prototype.retrieve = function (id, done) {
-  request(
-    {
-      url: this.uri + '/' + id,
-      auth: {
-        user: this.apiKey,
-        password: ''
-      },
-      headers: this.headers,
-      json: true,
-      method: 'GET'
-    }, function (e, r, body) {
-      if (body && body.errors) {
-        done(body.errors);
-      } else {
-        done(e, body);
-      }
-    });
-};
+'use strict';
 
-Areas.prototype.list = function (options, done) {
-  var count;
-  var offset;
-  if (typeof options === 'function') {
-    count = 10;
-    offset = 0;
-    done = options;
-  } else {
-    count = options.count || 10;
-    offset = options.offset || 0;
-  }
-
-  request(
-    {
-      url: this.uri,
-      auth: {
-        user: this.apiKey,
-        password: ''
-      },
-      qs: {
-        count: count,
-        offset: offset
-      },
-      headers: this.headers,
-      json: true,
-      method: 'GET'
-    }, function (e, r, body) {
-      if (body && body.errors) {
-        done(body.errors, body);
-      } else {
-        done(e, body);
-      }
-    });
+var util         = require('util');
+var fs           = require('fs');
+var ResourceBase = require('./resourceBase');
+
+var Areas = function(config) {
+  ResourceBase.call(this, 'areas', config);
 };
 
-Areas.prototype.create = function (params, done) {
-  var opts = {
-    url: this.uri,
-    auth: {
-      user: this.apiKey,
-      password: ''
-    },
-    headers: this.headers,
-    json: true
-  };
+util.inherits(Areas, ResourceBase);
+
+(function() {
 
-  var req = request.post(opts, function (e, r, body) {
-    if (body && body.errors) {
-      done(body.errors, body);
-    } else {
-      done(e, body);
+  this.list = function(options, callback) {
+
+    if(typeof options === 'function') {
+      callback = options;
+      options  = {};
     }
-  });
-  var form = req.form();
-  /* istanbul ignore else */
-  if (params.front) {
-    if (!(params.front instanceof Buffer) && params.front.substr(0,1) === '@') {
-      params.front = fs.createReadStream(params.front.substr(1));
-      form.append('front', params.front);
-    } else if (params.front instanceof Buffer) {
-      form.append('front', params.front, {filename: 'front.pdf'});
-    } else {
-      form.append('front', params.front);
+
+    options = {
+      count:  options.count  || 10,
```

Comment:
@leore is this the case for all resources which allow for count and offset (the server has the defaults? If so, I'll go through and update any resources which are setting these defaults and update them.